src/app/api/staff/mark-story/[staff_pick_id]/route.ts

```diff
@@ -0,0 +1,99 @@
+import prisma from "@/lib/prisma";
+import { getServerSession } from "next-auth";
+import { NextRequest, NextResponse } from "next/server";
+import { patchSchema } from "../schema";
+
+interface Params {
+  staff_pick_id: string;
+}
+export async function DELETE(req: NextRequest, { params }: { params: Params }) {
+  try {
+    const { staff_pick_id } = params;
+    const session = await getServerSession();
+    const user = await prisma.user.findUnique({
+      where: { email: session?.user.email ?? "noemail" },
+    });
+
+    if (!user || !user.roles.includes("EDITOR")) {
+      return NextResponse.json({ error: "Forbidden" }, { status: 403 });
+    }
+    const staffPickOld = await prisma.staffPick.findUnique({
+      where: { id: staff_pick_id },
+    });
+    if (!staffPickOld) {
+      return NextResponse.json({ error: "Not found" }, { status: 404 });
+    }
+
+    const deleteStaffPickPromise = prisma.staffPick.delete({
+      where: { id: staff_pick_id },
+    });
+    //create record
+    const createRecordPromise = prisma.staffPickRecord.create({
+      data: {
+        updateType: "DELETE",
+        storyId: staffPickOld.storyId,
+        staffId: user.id,
+      },
+    });
+    await Promise.all([deleteStaffPickPromise, createRecordPromise]);
+    return NextResponse.json({}, { status: 200 });
+  } catch (err) {
+    console.error(err);
+    return NextResponse.json(
+      { error: "Internal Server Error" },
+      { status: 500 },
+    );
+  }
+}
+
+export async function PATCH(
+  request: NextRequest,
+  { params }: { params: Params },
+) {
+  try {
+    const session = await getServerSession();
+    const user = await prisma.user.findUnique({
+      where: { email: session?.user.email ?? "noemail" },
+    });
+    const { staff_pick_id } = params;
+
+    if (!user || !user.roles.includes("EDITOR")) {
+      return NextResponse.json({ error: "Forbidden" }, { status: 403 });
+    }
+    const staffPickOld = await prisma.staffPick.findUnique({
+      where: { id: staff_pick_id },
+    });
+    if (!staffPickOld) {
+      return NextResponse.json({ error: "Not found" }, { status: 404 });
+    }
+    const { description } = patchSchema.parse(await request.json());
```

Comment:
When you parse like this, I think it throws an error if the item does not match the patchSchema.

it looks like this error will be caught by the final catch and return status 500

is it possible to split things a little, so when you catch this error, you return status 400 and the zod error?

Reply:
That is a good idea. I will try to figure out how to catch the schema and implement it.

Reply:
was this changed?

Reply:
> was this changed?

Yes, I changed to use the patchSchema.safeParse instead of parse. This will help us check the structure without throw the error